File: JS/NotasDaProva.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./NotasDaProva.js')

TEST('Starting NotasDaProva test...', (t) => {
    //Test 1
    t.assert(OBJ.takeResult(12) === "D", "Expect: 12 = D")
    //Test 2
    t.assert(OBJ.takeResult(87) === "A", "Expect: 87 = A")
    //Test 3
    t.assert(OBJ.takeResult(0) === "E", "Expect: 0 = E")

    t.end()
})

File: JAVA/NotasDaProvaTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class NotasDaProvaTest {

	@Test
	public void testNotasDaProvaTest() throws Exception {
		//Test 1
		String want = "D";
		String got = NotasDaProva.takeResult(12);
		assertEquals(want, got);
		//Test 2
		want = "A";
		got = NotasDaProva.takeResult(87);
		assertEquals(want, got);
		//Test 3
		want = "E";
		got = NotasDaProva.takeResult(0);
		assertEquals(want, got);
	}

}
